schema/2.0.0/corpora_schema.md

```diff
@@ -0,0 +1,257 @@
+    See this [list](https://docs.google.com/document/d/1sboOmbafvMh3VYjK1-3MAUt0I13UUJfkQseq8ANLPl8/edit) for guidance.
+
+#### *Note on types*
+The types below are python3 types. Note that a python3 `str` is a sequence of Unicode code points, which is stored null-terminated and UTF-8-encoded by anndata.
+
+## `X` (Matrix Layers)
+
+cellxgene does not impose any additional constraints on the `X` data matrix.
+
+In any layer, if a matrix has 50% or more  values that are zeros, it is STRONGLY RECOMMENDED that the matrix be encoded as a [`scipy.sparse.csr_matrix`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html).
+
+cellxgene's matrix layer requirements are tailored to optimize data reuse. Because each assay has different characteristics, the requirements differ by assay type. In general,
+cellxgene requires submission of "raw" data suitable for computational reuse when a standard raw matrix format exists for an assay and strongly recommends that a "final" matrix suitable for
+visualization in the explorer be included. So that cellxgene's data can be provided in download formats suitable for both R and Python, the schema imposes the following requirements:
+
+*   All matrix layers MUST have the same shape, and have the same cell labels and gene labels.
+*   Because it is impractical to retain all barcodes in raw and final matrices, any cell filtering MUST be applied to both.
+    By contrast, those wishing to reuse datasets require access to raw gene expression values, so genes MUST NOT be filtered from either dataset.
+    Summarizing, any cell barcodes that are removed from the data MUST be filtered from both raw and final matrices and genes MUST NOT be filtered from the raw matrix.
+*   Any genes that publishers wish to filter from the final matrix MAY have their values replaced by a language appropriate "null" value (e.g. [`np.nan`](https://numpy.org/doc/stable/reference/constants.html#numpy.nan) for python), which will mask them from exploration.
+*   Additional layers provided at author discretion MAY be stored using author-selected keys, but MUST have the same cells and genes as other layers.
+
+The following table describes the matrix data and layers requirements that are **assay-specific**. If an entry in the table is empty, the cellxgene schema does not have any other requirements on data in those layers beyond the ones listed above.
+
+| Assay | "raw" required? | "raw" requirements | "final" required? | "final" requirements | Other layers |
+|-|-|-|-|-|-|
+| scRNA-seq (UMI, e.g. 10x v3) | REQUIRED in `AnnData.raw.X` unless no "final" is provided, then `AnnData.X` | Values MUST be de-duplicated molecule counts. | STRONGLY RECOMMENDED |  | OPTIONAL |
+| scRNA-seq (non-UMI, e.g. SS2) | REQUIRED in `AnnData.raw.X` unless no "final" is provided, then `AnnData.X` | Values MUST be one of read counts (e.g. FeatureCounts) or  estimated fragments (e.g. output of RSEM). | STRONGLY RECOMMENDED |  | OPTIONAL |
```

Comment:
I agree that this could be useful to have somewhere, maybe in uns. However this type of information we don't use anywhere and we are not planning on using it anytime soon. I would lean towards not adding it in this iteration, as we have tried to keep it as minimal as possible to speed up the update of the validator and migration of data.

Reply:
@ambrosejcarr - is this something that you want to track for consideration in a future version of the schema?

Reply:
> @ambrosejcarr - is this something that you want to track for consideration in a future version of the schema?

No. I think it's fairly easily inferred from the matrix data type + assay type.